packages/code-du-travail-modeles/src/internal/ExtractSupportedCc.ts

```diff
@@ -1,9 +1,31 @@
 import Engine, { Rule } from "publicodes";
 
-type RuleNodeIdcc = Rule & { idcc: number };
+export type RuleNodeIdcc = Rule & {
+  cdtn: { idcc: number; "préavis-retraite"?: boolean };
+};
 
-export function extractImplementedCc(engine: Engine): Array<number> {
+export type AgreementInfo = {
+  idcc: number;
+  preavisRetraite: boolean;
+};
+
+export function extractImplementedCc(engine: Engine): AgreementInfo[] {
   return Object.values(engine.getParsedRules())
-    .flatMap((rule) => (rule.rawNode as RuleNodeIdcc)["idcc"])
-    .filter((item) => item !== undefined);
+    .flatMap((rule) => {
+      if (
+        (rule.rawNode as RuleNodeIdcc).cdtn &&
+        (rule.rawNode as RuleNodeIdcc).cdtn
+      ) {
+        const idcc = (rule.rawNode as RuleNodeIdcc).cdtn.idcc;
+        if (idcc) {
```

Comment:
C'est possible qu'il n'y ai pas d'idcc ? On a un test qui check ça non ? (peut etre que j'ai pas bien compris le code)

Reply:
Oui, c'est possible car on itère sur toutes les règles en fait. Du coup on cherche les règles qui ont un noeud cdtn avec un attribut idcc dedans. J'ai modifié le code pour que ce soit plus lisible et compréhensible.